pkg/storage/unified/resource/resource.proto

```diff
@@ -218,8 +218,9 @@ message ListOptions {
 }
 
 enum ResourceVersionMatch {
-  NotOlderThan = 0;
-  Exact = 1;
+  Unset = 0;
+  NotOlderThan = 1;
+  Exact = 2;
```

Comment:
This violates protobuf best practices by changing the numeric tag of an existing value (`NotOlderThan` from `0` to `1`) and reusing a previously assigned tag (`0`) for a new value (`Unset`). This leads to incompatible interpretations of serialized data between old and new versions.

```suggestion
  NotOlderThan = 0;
  Exact = 1;
  Unset = 2;
```

Reply:
Right. Now we come to a clash within protobuf best practices.

[It's standard to have 0 as some "unknown" or "catch-all" that is used whenever someone does not explicitly set a known-good value.](https://protobuf.dev/best-practices/dos-donts/#unspecified-enum) This practice wasn't followed here, so we risk the fact that NotOlderThan will be set as default value, which is wrong. 

On the other hand one shouldn't change existing definitions.

So I'm now at a big of a crossroads. Considering this is not released yet, do you think it's fine to break the protobuf definition here in order to align back with the standard.

Reply:
Made the changes to not break backwards compatibility.